smart-collections.go

```diff
@@ -0,0 +1,144 @@
+func (smartCollection SmartCollection) PublishedAt() time.Time {
+	return smartCollection.publishedAt
+}
+
+// PublishedScope is whether the smart collection is published to the Point of Sale channel.
+/*
+	Valid Values:
+	- web: The collection is published to the Online Store channel but not published to the Point of Sale channel.
+	- global: The collection is published to both the Online Store channel and the Point of Sale channel.
+*/
+func (smartCollection SmartCollection) PublishedScope() string {
+	return smartCollection.publishedScope
+}
+
+// SortOrder is the order in which products in the collection appear.
+/*
+	Valid values:
+	- alpha-asc: Alphabetically, in ascending order (A - Z).
+	- alpha-desc: Alphabetically, in descending order (Z - A).
+	- best-selling: By best-selling products.
+	- created: By date created, in ascending order (oldest - newest).
+	- created-desc: By date created, in descending order (newest - oldest).
+	- manual: In the order set manually by the shop owner.
+	- price-asc: By price, in ascending order (lowest - highest).
+	- price-desc: By price, in descending order (highest - lowest).
+*/
+func (smartCollection SmartCollection) SortOrder() string {
+	return smartCollection.sortOrder
+}
+
+// TemplateSuffix is the suffix of the liquid template being used.
+/*
+	For example, if the value is custom, then the collection is using the collection.custom.liquid template.
+	If the value is null, then the collection is using the default collection.liquid.
+*/
+func (smartCollection SmartCollection) TemplateSuffix() string {
+	return smartCollection.templateSuffix
+}
+
+// Title is the name of the smart collection. (limit: 255 characters)
+func (smartCollection SmartCollection) Title() string {
+	return smartCollection.title
+}
+
+// UpdatedAt is the date and time when the smart collection was last modified.
+func (smartCollection SmartCollection) UpdatedAt() time.Time {
+	return smartCollection.updatedAt
+}
+
+// NewSmartCollection create new SmartColllection
+func NewSmartCollection(
+	bodyHTML string,
+	handle string,
+	id int64,
+	image Image,
+	productCount int,
+	publishedAt time.Time,
+	publishedScope string,
+	rules Rules,
+	disjunctive bool,
+	sortOrder string,
+	templateSuffix string,
+	title string,
+	updatedAt time.Time,
+) SmartCollection {
+	return SmartCollection{
+		bodyHTML:       bodyHTML,
+		handle:         handle,
+		id:             id,
+		image:          image,
+		productCount:   productCount,
+		publishedAt:    publishedAt,
+		publishedScope: publishedScope,
+		Rules:          rules,
+		Disjunctive:    disjunctive,
+		sortOrder:      sortOrder,
+		templateSuffix: templateSuffix,
+		title:          title,
+		updatedAt:      updatedAt,
+	}
+}
+
+// SmartCollectionRepository maintains the collections of a shop.
+type SmartCollectionRepository interface {
+	// Get retrieves a single smart collection
+	Get(id int64) (SmartCollection, error)
+}
```

Comment:
delete this and the custom repo one if we arent going to link them to the shop now